src/events/processSignatureRequests.js

```diff
@@ -2,12 +2,65 @@ require('dotenv').config()
 const Web3 = require('web3')
 const HttpListProvider = require('http-list-provider')
 const promiseLimit = require('promise-limit')
+const bridgeValidatorsABI = require('../../abis/BridgeValidators.abi')
 const logger = require('../services/logger')
 const rpcUrlsManager = require('../services/getRpcUrlsManager')
 const { createMessage } = require('../utils/message')
+const {
+  AlreadyProcessedError,
+  AlreadySignedError,
+  InvalidValidatorError
+} = require('../utils/errors')
 const { MAX_CONCURRENT_EVENTS } = require('../utils/constants')
 
 const { VALIDATOR_ADDRESS, VALIDATOR_ADDRESS_PRIVATE_KEY } = process.env
+const { HttpListProviderError } = HttpListProvider
+
+async function estimateGas(web3, bridgeContract, signature, message, address) {
+  try {
+    const gasEstimate = await bridgeContract.methods
+      .submitSignature(signature, message)
+      .estimateGas({
+        from: address
+      })
+    return gasEstimate
+  } catch (e) {
+    if (e instanceof HttpListProviderError) {
+      throw e
+    }
+
+    // Check if address is validator
```

Comment:
@akolotov The order of this check is not the most efficient. They should be sorted from most likely to less likely. I think a better order would be:

1) Check if the tx was already processed by the other validators
2) Check if the tx was already signed by this validator
3) Check if the validator is still in the list of validators.

Do you agree?